lib/auth/auth_with_roles.go

```diff
@@ -3687,6 +3701,66 @@ func checkRoleFeatureSupport(role types.Role) error {
 	}
 }
 
+type inventoryGetter func(context.Context, proto.InventoryStatusRequest) proto.InventoryStatusSummary
+
+// checkInventorySupportsRole returns an error if any connected servers found in
+// the inventory do not support some features enabled in [role]. This is only a
+// best-effort check meant to prevent common user errors, since some unsupported
+// servers may not be connected to this auth, or they may connect later.
+func checkInventorySupportsRole(ctx context.Context, role types.Role, authVersion string, getInventory inventoryGetter) error {
+	minRequiredVersion, msg, err := minRequiredVersionForRole(role)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if semver.New(authVersion).Major > minRequiredVersion.Major+1 {
+		// The version of this auth server is more than one major version ahead
+		// of the minimum required version for the role, so any servers older
+		// than minRequiredVersion are already unsupported and shouldn't be
+		// connected. Skip the inventory check in this case as an optimization.
+		// This also covers the case where minRequiredVersionForRole returned
+		// the zero version. This effectively means we will automatically stop
+		// doing the inventory checks in v15.
+		return nil
+	}
```

Comment:
I think you're off by one, this checks if the auth server version is *greater than* `minRequiredVersion.Major+1`, we won't take the early return here until 15.0.0-alpha

Reply:
Ah, you're right... totally misunderstood this check.

Actually, now that I'm re-reading it... I suppose this check actually risks failing in the exact opposite way of what I originally thought lol (not for this specific feature, but for other ones)... shouldn't we be "rounding" up rather than just incrementing... i.e. `v13.0.1` should become `v14`, but `v13.0.0` should just be `v13`... right?  That way we also correctly handle the case of features there *were* introduced in a major version?  NBD either way, I'm always happy to err on the side of caution.

Reply:
oh you're totally right. It should be handled properly in the latest, added some testcases to be sure